src/rust/engine/options/src/arg_splitter.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2025 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::collections::HashSet;
+use std::path::{Path, PathBuf};
+
+lazy_static! {
+    static ref SPEC_RE: Regex = Regex::new(r"[/\\.:*#]").unwrap();
+    static ref SINGLE_DASH_FLAGS: HashSet<&'static str> =
+        HashSet::from(["-ltrace", "-ldebug", "-linfo", "-lwarn", "-lerror", "-h", "-v", "-V"]);
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct SplitArgs {
+    pub goals: Vec<String>,         // The requested known goals.
+    pub unknown_goals: Vec<String>, // Any requested but unknown goals.
+    pub specs: Vec<String>,         // What to run against, e.g. targets or files/dirs.
+    pub passthru: Vec<String>,      // Any remaining args specified after a -- separator.
+}
+
+pub struct ArgSplitter {
+    build_root: PathBuf,
+    known_goal_names: HashSet<String>,
+}
+
+impl ArgSplitter {
+    pub fn new<'a, I: IntoIterator<Item = &'a str>>(
+        build_root: &Path,
+        known_goal_names: I,
+    ) -> ArgSplitter {
+        ArgSplitter {
+            build_root: build_root.to_owned(),
+            known_goal_names: HashSet::from_iter(known_goal_names.into_iter().map(str::to_string)),
```

Comment:
More idiomatic might be to use `collect`:  `known_goal_names.into_iter().map(str::to_string).collect()`

Rust should infer the type from the type of the `known_goal_names` field.

Reply:
Good idea. Done.